src/PSRule.Rules.Azure/rules/Azure.MySQL.Rule.ps1

```diff
@@ -87,6 +87,23 @@ Rule 'Azure.MySQL.AAD' -Ref 'AZR-000392' -Type 'Microsoft.DBforMySQL/flexibleSer
     }
 }
 
+# Synopsis: Ensure Azure AD-only authentication is enabled with Azure Database for MySQL databases.    
+Rule 'Azure.MySQL.AADOnly' -Ref 'AZR-000394' -Type 'Microsoft.DBforMySQL/flexibleServers', 'Microsoft.DBforMySQL/flexibleServers/configurations' -If { IsExport } -Tag @{ release = 'GA'; ruleSet = '2023_09'; 'Azure.WAF/pillar' = 'Security'; } -Labels @{ 'Azure.MCSB.v1/control' = 'IM-1' } {
+    if ($PSRule.TargetType -eq 'Microsoft.DBforMySQL/flexibleServers') {
+        $configurations = @(GetSubResources -ResourceType 'Microsoft.DBforMySQL/flexibleServers/configurations' -Name "$($PSRule.TargetName)/aad_auth_only")
```

Comment:
Try this. The function does a split on the name and only looks for resource by their last segment.

```suggestion
        $configurations = @(GetSubResources -ResourceType 'Microsoft.DBforMySQL/flexibleServers/configurations' -Name "aad_auth_only")
```

Reply:
@BernieWhite Found this out, works as it should. Thanks! Mistake from my side, I forgot that that the structure is applicable when it's outside the parent resource and the function is for within the parent resource, where the name values are as a single segment without slashes.